x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -58,6 +59,10 @@ export const postActionsConnectorExecuteRoute = (
         // TODO: Add `traceId` to actions request when calling via langchain
         logger.debug('Executing via langchain, assistantLangChain: true');
 
+        // Fetch any tools registered by the request's originating plugin
+        const pluginName = getPluginNameFromRequest({ request, logger });
```

Comment:
Nice solution to ensure `getRegisteredTools()` returns tools scoped to the plugin! 🎉